extensions/ql-vscode/src/variant-analysis/view-autofixes.ts

```diff
@@ -0,0 +1,801 @@
+      repoAutofixOutputStoragePath,
+    );
+  }
+
+  // Format the output text file with markdown.
+  await formatWithMarkdown(outputTextFilePath, `${nwo}`);
+
+  // Save output text files from each repo to later merge
+  // into a single markdown file containing all results.
+  outputTextFiles.push(outputTextFilePath);
+}
+
+/**
+ * Gets the storage paths for the autofix results for a given repository.
+ */
+async function getRepoStoragePaths(
+  autofixOutputStoragePath: string,
+  nwo: string,
+) {
+  // Create output directories for repo's autofix results.
+  const repoAutofixOutputStoragePath = join(
+    autofixOutputStoragePath,
+    nwo.replaceAll("/", "-"),
+  );
+  await ensureDir(repoAutofixOutputStoragePath);
+  return {
+    repoAutofixOutputStoragePath,
+    outputTextFilePath: join(repoAutofixOutputStoragePath, "output.txt"),
+    transcriptFilePath: join(repoAutofixOutputStoragePath, "transcript.md"),
+    fixDescriptionFilePath: join(
+      repoAutofixOutputStoragePath,
+      "fix-description.md",
+    ),
+  };
+}
+
+/**
+ * Runs autofix on the results in the given SARIF file.
+ */
+async function runAutofixOnResults(
+  logger: NotificationLogger,
+  cocofixBin: string,
+  sarifFile: string,
+  srcRootPath: string,
+  outputTextFilePath: string,
+  fixDescriptionFilePath: string,
+  transcriptFilePath: string,
+  workDir: string,
+  alertNumber?: number, // Optional parameter for specific alert
+): Promise<void> {
+  // Set up args for autofix command.
+  const fixedAutofixArgs = [
+    "--sarif",
+    sarifFile,
+    "--source-root",
+    srcRootPath,
+    "--model",
+    "capi-dev-4o", // may fail with older versions of cocofix
+    "--dev",
+    "--no-cache",
+    "--format",
+    "text",
+    "--diff-style",
+    "diff", // could do "text" instead if want line of "=" between fixes
+  ];
+  const varAutofixArgs = createVarAutofixArgs(
+    outputTextFilePath,
+    fixDescriptionFilePath,
+    transcriptFilePath,
+    alertNumber,
+  );
+
+  const autofixArgs = [...fixedAutofixArgs, ...varAutofixArgs];
+
+  await execAutofix(
+    logger,
+    cocofixBin,
+    autofixArgs,
+    {
+      cwd: workDir,
+      env: {
+        CAPI_DEV_KEY: process.env.CAPI_DEV_KEY,
+        PATH: process.env.PATH,
+      },
+    },
+    true,
+  );
+}
+
+/**
+ * Creates autofix arguments that vary depending on the run.
+ */
+function createVarAutofixArgs(
+  outputTextFilePath: string,
+  fixDescriptionFilePath: string,
+  transcriptFilePath: string,
+  alertNumber?: number, // Optional parameter for specific alert
+): string[] {
+  const args = [
+    "--output",
+    outputTextFilePath,
+    "--fix-description",
+    fixDescriptionFilePath,
+    "--transcript",
+    transcriptFilePath,
+  ];
+
+  // Add alert number argument if provided
+  if (alertNumber !== undefined) {
+    args.push("--only-alert-number", alertNumber.toString());
+  }
+
+  return args;
+}
+
+/**
+ * Executes the autofix command.
+ */
+function execAutofix(
```

Comment:
Note: this function is mostly the same as DCA's `execFollow` function which DCA uses to execute autofix.